core/src/main/scripts/migrate_db.py

```diff
@@ -29,86 +29,63 @@
 class PortalProperties(object):
     """ Properties object class, just has fields for db conn """
 
-    def __init__(self, database_host, database_name, database_user, database_pw, database_use_ssl):
-        # default port:
-        self.database_port = 3306
-        # if there is a port added to the host name, split and use this one:
-        if ':' in database_host:
-            host_and_port = database_host.split(':')
-            self.database_host = host_and_port[0]
-            if self.database_host.strip() == 'localhost':
-                print(
-                    "Invalid host config '" + database_host + "' in properties file. If you want to specify a port on local host use '127.0.0.1' instead of 'localhost'",
-                    file=ERROR_FILE)
-                sys.exit(1)
-            self.database_port = int(host_and_port[1])
-        else:
-            self.database_host = database_host
-        self.database_name = database_name
+    def __init__(self, database_user, database_pw, database_url):
         self.database_user = database_user
         self.database_pw = database_pw
-        self.database_use_ssl = database_use_ssl
+        self.database_url = database_url
 
 def get_db_cursor(portal_properties):
     """ Establishes a MySQL connection """
     try:
-        connection_kwargs = {}
-        connection_kwargs['host'] = portal_properties.database_host
-        connection_kwargs['port'] = portal_properties.database_port
-        connection_kwargs['user'] = portal_properties.database_user
-        connection_kwargs['passwd'] = portal_properties.database_pw
-        connection_kwargs['db'] = portal_properties.database_name
-        if portal_properties.database_use_ssl == 'true':
-            connection_kwargs['ssl'] = {"ssl_mode": True}
-        
+        url_elements = dsnparse.parse(portal_properties.database_url)
+        connection_kwargs = {
+            "host": url_elements.host,
+            "port": url_elements.port if url_elements.port is not None else 3306,
+            "db": url_elements.paths[0],
+            "user": portal_properties.database_user,
+            "passwd": portal_properties.database_pw,
+            "ssl": "useSSL" not in url_elements.query or url_elements.query["useSSL"] == "true"
+        }
```

Comment:
I see we parsed `database_url` here to get all DB properties. But it seems like the `database_url` is required and not backward compatible. In the description of this pr, I see `This implementation is backward compatible with configuring with the legacy db.host, db.database, db.port and db.use_ssl properties.`. So I was expecting to see it's fully compatible with current DB properties. Are there any reasons for making `database_url` a required property here?